### PR TITLE
Stabilizing eden integration tests

### PR DESCRIPTION
**Pull request by milan-zededa**
This PR contains 3 commits that are addressing at least some issues that we are frequently seeing in the eden integration testing.

**Make domainmgr skip DNS with empty DPCKey**

Do not activate `PhysicalIOAdapterList` subscription until NIM receives
a DPC and publishes corresponding DNS.
NIM can publish DNS even before it receives first DPC. In such case
`DPCKey` is empty.
The goal is to avoid assigning network ports to PCIBack if they are going to be
used for management purposes. This way we avoid doing unintended port assignment
to PCIBack that would be shortly followed by a release, therefore mitigating
the risk of race conditions between domainmgr and NIM.

**Increase timeout for containerd client connection**

This is mostly to address error frequently seen when run eden tests on
(slow) github runners:
```
Rebooting EVE. Reason: Reboot from agent zedbox[1468] in partition IMGA at EVE version pr3184-kvm-amd64 at 2023-05-02T23:53:59.207011689Z: fatal: agent zedbox[1468]: couldn't initialize containerd (this should not happen): initContainerdClient: could not create containerd client. failed to dial "/run/containerd/containerd.sock": context deadline exceeded: failed to receive server preface within timeout. Exiting.
```
I have not seen this when running tests locally or when running EVE
on bare-metal devices.
Note that default timeout for containerd client to connect is 10
seconds. Increasing this to 30 seconds should not have any negative
unintended consequences.

**Update eden tests**

Bring in patches done in the eden repository to address some
unstable eden tests.

### CODE REVIEW

File: .github/workflows/eden.yml [hunk 1]
```diff
@@ -48,7 +48,7 @@ jobs:
           if [ -f ${{ github.workspace }}/eve/tests/eden/eden-version ]; then
             EDEN_VERSION=$(cat ${{ github.workspace }}/eve/tests/eden/eden-version)
           else
-            EDEN_VERSION=lfedge/eden:0.8.0
```
Comment (eriknordmark) [inline .github/workflows/eden.yml:51]: My understanding from @giggsoff a long time ago is that this is just a fallback and that we should update /eve/tests/eden/eden-version when we want a new version.

File: pkg/pillar/cmd/domainmgr/domainmgr.go [hunk 2]
```diff
@@ -2410,6 +2410,19 @@ func handleDNSImpl(ctxArg interface{}, key string,
...
+		// Do not activate PhysicalIOAdapterList subscription until NIM receives
+		// a DPC and publishes corresponding DNS.
+		// NIM can publish DNS even before it receives first DPC. In such case
+		// DPCKey is empty.
+		// The goal is to avoid assigning network ports to PCIBack if they are going to be
+		// used for management purposes. This way we avoid doing unintended port assignment
+		// to PCIBack that would be shortly followed by a release, therefore mitigating
+		// the risk of race conditions between domainmgr and NIM.
+		log.Warnf("handleDNSImpl: DNS with empty DPCKey")
+		ctx.DNSinitialized = false
```
Comment (eriknordmark) [inline pkg/pillar/cmd/domainmgr/domainmgr.go:2423]: Are there flows were we set DNSinitialized to true and then later end up setting it to false here?
I'm trying to understand any interaction with any wait for DNSinitialized in Run.

**Review by eriknordmark — APPROVED** [review]
Kick off tests.

File: pkg/pillar/cmd/domainmgr/domainmgr.go (hunk 2 above)
Comment (milan-zededa) [reply pkg/pillar/cmd/domainmgr/domainmgr.go:2423]: domainmgr enable PhysicalIO subscription when `DNSinitialized` changes from `false` to `true`. The other way it has no effect. Actually, I could remove `ctx.DNSinitialized = false` and just return.

File: .github/workflows/eden.yml (hunk 1 above)
Comment (milan-zededa) [reply .github/workflows/eden.yml]: I did change both `eden-version` and these fallbacks. But now I decided to revert change for fallbacks - let's do that later when we will make next eden tag.

**Review by uncleDecart — APPROVED** [review]
LGTM

**Comment by milan-zededa** [comment]
The only fail that we caught here is something that I'm aware of but couldn't yet find the root cause.
It is some mysterious reboot that EVE thinks is kernel crash but there is no kdump. Happens fairly often in `dev_local_info` test.
```
        === RUN   TestReboot
            reboot_test.go:162: time: 2023-05-14T23:00:56.920564214Z out: Wait for state of 1e7471c7-016d-4cfb-91b3-0428d370159c
            testContext.go:380: done waiting for State
            reboot_test.go:166: time: 2023-05-14T23:00:56.920591815Z out: timewait: 1h40m0s
            reboot_test.go:167: time: 2023-05-14T23:00:56.920598515Z out: reboot: false
            reboot_test.go:168: time: 2023-05-14T23:00:56.920603715Z out: count: 3
            reboot_test.go:172: time: 2023-05-14T23:00:56.920611515Z out: LastRebootTime: 2023-05-14 19:12:29.340669651 +0000 UTC
            reboot_test.go:174: time: 2023-05-14T23:00:56.920617615Z out: LastRebootReason: NORMAL: First boot of device - at 2023-05-14T19:12:29.340669651Z
        rebooted with reason NORMAL: controller reboot at EVE version 0.0.0-pr3208-c0d54d46-kvm-amd64 at 2023-05-14T23:15:50.070481583Z
         at 2023-05-14 23:15:50.149733256 +0000 UTC/nrebooted with reason NORMAL: local profile server power off at EVE version 0.0.0-pr3208-c0d54d46-kvm-amd64 at 2023-05-14T23:27:34.443011196Z
         at 2023-05-14 23:27:34.964623967 +0000 UTC/nrebooted with reason Reboot reason - kernel crash (no kdump) - at 2023-05-14T23:40:08.888905598Z at 2023-05-14 23:40:08.888905598 +0000 UTC/n    testProc.go:84: time: 2023-05-14T23:41:33.675024866Z out: projects.ProcInfoFunc done with return: abnormal reboot: Reboot reason - kernel crash (no kdump) - at 2023-05-14T23:40:08.888905598Z
            testContext.go:271: done for device 1e7471c7-016d-4cfb-91b3-0428d370159c
            reboot_test.go:186: time: 2023-05-14T23:41:33.675099367Z out: Number of reboots: 2
                
        --- PASS: TestReboot (2436.75s)
        PASS
        [stderr]
        time="2023-05-14T23:41:33Z" level=info msg="time: 2023-05-14T23:41:33.675024866Z out: projects.ProcInfoFunc done with return: abnormal reboot: Reboot reason - kernel crash (no kdump) - at 2023-05-14T23:40:08.888905598Z"
```

**Pull request merged**